solidity/contracts/BeaconRewards.sol

```diff
@@ -0,0 +1,84 @@
+pragma solidity ^0.5.17;
+
+import "./Rewards.sol";
+import "./KeepRandomBeaconOperator.sol";
+import "./TokenStaking.sol";
+
+contract BeaconRewards is Rewards {
+    KeepRandomBeaconOperator operatorContract;
+    TokenStaking tokenStaking;
+
+    constructor (
+        uint256 _termLength,
+        address _token,
+        uint256 _minimumKeepsPerInterval,
+        uint256 _firstIntervalStart,
+        uint256[] memory _intervalWeights,
+        address _operatorContract,
+        address _stakingContract
+    ) public Rewards(
+        _termLength,
+        _token,
+        _minimumKeepsPerInterval,
+        _firstIntervalStart,
+        _intervalWeights
+    ) {
+        operatorContract = KeepRandomBeaconOperator(_operatorContract);
+        tokenStaking = TokenStaking(_stakingContract);
+    }
+
+    function _getKeepCount() internal view returns (uint256) {
+        return operatorContract.getNumberOfCreatedGroups();
```

Comment:
How are we going to deal with the existing operator contract deployed where this function is not available?

Reply:
I made a couple of mistakes in getting the operator contract rewards-compatible and didn't realize it until after #1851 was merged. It's possible to hunt for the group number by binary searching the highest registered group index, but the existing operator contract is also missing the registration timestamp storage, meaning that to properly support it the entire rewards contract would need to be adapted to use either timestamps or blockheights depending on configuration. It can be done, but it would be all around easier if we could deploy the operator contract with these changes.

Reply:
Yeah, I remember the previous proposition used the binary search and that's why I was confused. Long story short, are we good with the operator contract after the changes from #1851, or do we need some additional tweaks there?

If making the rewards contract support the current mainnet version of the operator contract and bonded ecdsa keep factory is painful, we can maybe agree on distributing KEEP rewards to existing stakers manually, with a separate contract. @mhluongo what do you think about it?

Reply:
> we can maybe agree on distributing KEEP rewards to existing stakers semi-manually, with a separate contract. @mhluongo what do you think about it?

If semi-manually means we take custody, nope nope nope. But if you mean we are the ones okaying distributions... maybe? Definitely want to avoid it if possible.

Reply:
No, I don't want us to take custody. I was rather thinking about having a clone of this contract adjusted to the existing operator contracts with some of the functionality bypassed. For example, we can hardcode the number of groups created and their public keys.

I don't know what's the cost of making such a separate contract versus the cost of adjusting this contract to work with the currently deployed versions. For sure, the rewards contract as-is does not support them.

Reply:
We have `BeaconBackportRewards` now. Resolving this conversation.